### ISSUE TITLE
Reorganize repo to be "mdn/content-linter"?

### ISSUE THREAD

**Issue by Elchi3** [mdn/stumptown-content#468]
I'm sorry to file this right after sprint planning and please take it as a suggestion not as a thing I want us to do right now necessarily. I think it would be nice to discuss it here, though. 

For two reasons Will mentioned in his June stumptown summary, I'd like us to consider framing our work as "mdn/content-linter".

> We've found and fixed many errors to get to this point. And now we can track these assertions, so we know when, for example, new pages are added that break them. When our docs live in GitHub we should be able to run the linter in CI, so we fail the builds if PRs break our assertions.

I ran the content linter against JS docs today and turns out someone wrote new docs yesterday and (no surprise) they don't conform to the linting rules, because "the linter" isn't a thing that is available to contributors. So, I believe, if our goal is to have zero linting errors in JS docs not just this week, but going forward, we need to make the linter a bit more accessible than it is today.

For when our content lives in GitHub and the linter should check it as part of CI, it probably needs to become a proper organized package anyway.

I think, by making linter more accessible, we will also receive more feedback and probably even more contributions, resulting in more or in better lints.

Thoughts?

**Comment by ddbeck** [comment]
I completely agree with this. We need to make it plausible for others to use the linter and giving it a more prominent place in a repo makes sense.

Plus, the linter itself has become rather overgrown as a "script" in stumptown-content. If we expect people to use it, we probably also expect that people will contribute to it. Right now, that's harder than it needs to be, given its subordinate position in the stumptown package.

There are a couple of hurdles though. The biggest, to me, is recipes. stumptown-content needs them and the linter needs them. It makes sense that the recipes live with stumptown-content and it makes sense that the recipes live with the linter (we've more or less specified the linter with that understanding). I'm not sure how to extricate the content linter in a way that doesn't signal abandonment of the rest of the repo (though I suppose we've already done that to some degree).

**Comment by wbamberg** [comment]
I don't care one bit about the build-json code in this repo which converts structured Markdown into JSON. But if we are still interested in representing our content as structured Markdown, then I do care about the structures we have defined: both the Markdown+YAML we have for the authoring interface and the JSON we have for the consumer interface.

I agree that the way this repo is organized at the moment is terribly obscure. But we also need to think about the future as well as our current needs. Right now, linting HTML+KS pages is all we are doing. But we had thought of this as preparation for scraping: that is transforming HTML+KS pages into structured Markdown. Is that still the idea? If so, when we do that (if we do that) then what will we want this repo to be named, and what will we want it to contain? Are we going to rename it again? Or is it going to be a separate repo?

Also we should consider that "content-linter" is a tricky name. The current linter lints pages in their HTML+KS form. It's a "wiki-content-linter", for want of a better name. If we ever get into structured Markdown, we need a new linter, which will be quite different from this one (it won't have to examine markup to nearly the same extent).

**Comment by ddbeck** [comment]
After thinking in circles about this, I don't think I can solve the repo name problem in a satisfactory way. If we break up the repo, then I think we'd just end up with a bunch of names that obscure the relationships (e.g., mdn/recipes, mdn/wiki-content-linter, mdn/stumptown, etc.), which is probably worse. Plus it signals a foreclosure on scraping (or at least creates a different version of the naming problem down the road).

But I think we can reorganize the repo to be more understandable and maintainable, which is probably far more critical to making this repo approachable to outside users and contributors than the repo name. There's a few pieces to this, but I think the main things are:

1. Promote the recipes and linter to _be_ the `stumptown` package (e.g., you'd have submodules like `stumptown.recipes`, `stumptown.fetch`, `stumptown.lint`, `stumptown.cli`, `stumptown.test`).
2. Demote `build-json` etc. (the current `stump` package) to an `experiments` folder (or delete it).

Breaking out the `scripts` strucure would free me up to do lots of things that would make the linter and the recipes more comfortable to use and help us not be an island. For instance, right now I don't think we could expose an API to Yari (for its flaws system) or offer an alternative command-line interface to run the linter on local files.

But I'm also not quite sure how to structure this work. There's some low-hanging fruit I could just open PRs for, but I'd feel more confident about this if it were more intentional and planned than that. But perhaps that's out of scope for this issue.
